codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowControllerTest.java

```diff
@@ -1050,6 +1050,10 @@ public void error(Throwable t) {
         }
 
         @Override
+        public void writeComplete() {
```

Comment:
Some ideas for testing the flow controller portion of the logic:

1. Call a mocked `FlowControlled.write()` which writes a little and then throws. Verify that flow control windows are updated properly.
2. Same test as 1, but also throw from `error()` and verify that the flow control windows are still updated properly.
3. Throw from `complete()` and verify the flow control windows.
4. In the mocked `error` method, close the stream and verify that `error` is only called the one time for that pending write.
5. In the mocked `complete()` method, close the stream and verify that `error` is not called and flow control is updated properly.

Any other ideas for things to test?